src/LinkedListVisualizer/LinkedListVisualizer.jsx

```diff
@@ -0,0 +1,497 @@
+                    Clear Board
+            </button>
+           
+            </center>
+
+            <div className="pushLeft">
+            <div className="flexRow">
+                {grid.map((node) => {
+                    const {location, data, nextNode} = node;
+                    
+                    if(location === (numberOfNodes - 1) && this.state.cyclePresent){
+                        return(
+                            <div>
+
+                            <div className="flexRow">
+                            <Node 
+                            location={location}
+                            data={data}
+                            key={location}
+                            nextNode={nextNode}
+                            />
+                            </div>
+                            
+                            <div style={{height: 100, backgroundColor: "#435058", width: this.state.downArrowWidth, position: "relative", left: (this.state.nodeWidth - this.state.downArrowWidth) / 2 }}>
+                            </div>
+
+                            </div>
+                        )
+                    } else if(location == this.state.cycleTo && this.state.cyclePresent){
+                        return(
+                            <div>
+
+                            <div className="flexRow">
+                            <Node 
+                            location={location}
+                            data={data}
+                            key={location}
+                            nextNode={nextNode}
+                            />
+                            <Pointer></Pointer>
+
+                            </div>
+                            <div style={{position: "relative", left: 10}}>
+                                <div className="upTriangle">
+                                </div>
+                            </div>
+
+                            <div style={{height: 100 - this.state.downArrowWidth, backgroundColor: "#435058", width: this.state.downArrowWidth, position: "relative", left: (this.state.nodeWidth - this.state.downArrowWidth) / 2 }}>
+                            </div>
+                            </div>
+                            )} else{
+                                    return(
+                                        <div>
+                            <div className="flexRow">
+                            <Node 
+                            location={location}
+                            data={data}
+                            key={location}
+                            nextNode={nextNode}
+                            />
+                            
+                            <Pointer></Pointer>
+                            </div>
+                            </div>
+                        )
+                    }
+                }
+                )}
+            </div>
+
+            <div>
+            {grid.map((node) => {
+                const {location} = node;
```

Comment:
Isn't `location` the same as the index in `grid`? Do you really need to go through all the nodes to do this?